Automated Smoketest: :x:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-88.ca-central-1.compute.internal
Sha: 0c1dc184a68a2bea82adb233c1aa310e10dd25f1
Number of Error(s): 7
warn(226) A package without a profile could throw errors or warnings. package: dojo-util
warn(226) A package without a profile could throw errors or warnings. package: dojo-util
\terrors: 0
c++: error: NOTFOUND: No such file or directory
make[2]: *** [ecl/hql/CMakeFiles/hql.dir/build.make:491: ecl/hql/CMakeFiles/hql.dir/hqlgram.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:8432: ecl/hql/CMakeFiles/hql.dir/all] Error 2
make: *** [Makefile:163: all] Error 2
Get ECLWatch build result
ReportTimes.
Finish at: 2020-09-07_14-13-00
time logs started
Prep time   :     9 sec (00:00:09)
Build time  :   508 sec (00:08:28)
Package time:     0 sec (00:00:00)
Install time:     0 sec (00:00:00)
Start time  :     0 sec (00:00:00)
Test time   :     0 sec (00:00:00)
Stop time   :     0 sec (00:00:00)
Summary     :   517 sec (00:08:37)
time logs end

[Build error:](https://gist.github.com/179bdf2b0f46ce09ab62384708ade4e4)
Build log [Build log](https://gist.github.com/HPCCSmoketest/179bdf2b0f46ce09ab62384708ade4e4#file-result-20-09-07-14-04-10-log)

--- a/ecl/hql/CMakeLists.txt
+++ b/ecl/hql/CMakeLists.txt
@@ -156,9 +156,9 @@ if (CMAKE_COMPILER_IS_GNUCC OR CMAKE_COMPILER_IS_CLANG)
     set_source_files_properties (hqlfilter.cpp PROPERTIES COMPILE_FLAGS " -Wimplicit-fallthrough=0")
   endif()
 
-  #Compiler complains about a memcpy of a structure that contains a union
+  #Compiler complains about a memcpy of a structure that contains a union, and copy if uninitialised enum
   if (CMAKE_COMPILER_IS_GNUCC AND NOT CMAKE_CXX_COMPILER_VERSION VERSION_LESS "8.0.0")
-    set_source_files_properties (hqlgram.cpp PROPERTIES COMPILE_FLAGS " -Wno-class-memaccess")
+    set_source_files_properties (hqlgram.cpp PROPERTIES COMPILE_FLAGS " -Wno-class-memaccess -fno-sanitize=undefined")
     set_source_files_properties (hqllex.cpp PROPERTIES COMPILE_FLAGS " -Wno-class-memaccess -Wno-sign-compare")
   endif()
 

